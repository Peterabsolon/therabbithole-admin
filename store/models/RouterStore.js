import { types } from 'mobx-state-tree'

export const RouterStore = types.model(
  'Router',
  {
    pathname: types.optional(types.string, ''),
    params: types.optional(types.string, ''),
  },
  {
    openContactPage() {
      this.pathname = '/contact'
    },
    init(pathname, params = '') {
      this.pathname = pathname
      this.params = params
    },
  }
)
