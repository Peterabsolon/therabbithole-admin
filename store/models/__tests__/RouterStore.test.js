// import { when } from 'mobx'
import { RouterStore } from '../RouterStore'

// const bookFetcher = () => Promise.resolve(JSON.parse(fs.readFileSync("./public/books.json")))

// it('bookstore fetches data', () => {
//   const store = AppStore.create({ title: 'App title' })
//   console.log('created store', store.sourcesStore.sources.length)
//   expect(store.sourcesStore.sources.length).toBe(0)
//   //   when(
//   //     () => store.isLoading === false,
//   //     () => {
//   //       expect(store.sourcesStore.sources.length).toBe(0)
//   //       done()
//   //     }
//   //   )
// })
describe('ðŸ—„ Router Store', () => {
  it('Init', () => {
    const store = RouterStore.create()
    expect(store).toMatchSnapshot()
  })

  it('Create with pathname', () => {
    const store = RouterStore.create({ pathname: '/test' })
    expect(store).toMatchSnapshot()
  })

  it('Open page', () => {
    const store = RouterStore.create({ pathname: '/test' })
    store.openContactPage()
    expect(store).toMatchSnapshot()
  })
})
