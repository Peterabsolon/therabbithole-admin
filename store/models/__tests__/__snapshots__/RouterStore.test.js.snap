// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ—„ Router Store Create with pathname 1`] = `
Object {
  "params": "",
  "pathname": "/test",
}
`;

exports[`ðŸ—„ Router Store Init 1`] = `
Object {
  "params": "",
  "pathname": "",
}
`;

exports[`ðŸ—„ Router Store Open page 1`] = `
Object {
  "params": "",
  "pathname": "/contact",
}
`;
